swap_meet/vendor.py

```diff
@@ -1,2 +1,138 @@
+        '''
+        for product in self.inventory:
+            if product == item:
+                self.inventory.remove(item)
+                return product
+        
+        return False
+
+    def get_by_category(self, category):
+        '''
+        appends items from inventory that match the category parameter
+        '''
+        item_list = []
+
+        for item in self.inventory:
+            if item.category == category:
+                item_list.append(item)
+        
+        return item_list
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        '''
+        swaps items between self and other inventories
+        '''
+        # Helpers
+        self_inventory = self.inventory 
+        other_inventory = other_vendor.inventory
+
+        if my_item not in self_inventory or their_item not in other_inventory or len(other_inventory) == 0 or len(self_inventory) == 0:
+            return False
+        
+        for item in self_inventory:
+            if item == my_item:
+                other_vendor.add(item)
+                self.remove(item)
+
+        for item in other_inventory:
+            if item == their_item:
+                self.add(item)
+                other_vendor.remove(item)
+        
+        return True
+
+    def swap_first_item(self, other_vendor):
+        '''
+        swaps the first items of self and other inventories
+        '''
+        # Swap 1st items
+        try:
+            # Helpers
+            self_inventory_item = self.inventory[0]
+            other_inventory_item = other_vendor.inventory[0]
+            return self.swap_items(other_vendor, self_inventory_item, other_inventory_item)
+
+        except:
+            return False
+
+    def get_best_by_category(self, category):
+        '''
+        returns the best condition item by category
+        '''
+        result = []
+
+        for item in self.inventory:
+            if item.category == category:
+                result.append(item)
+        
+        if len(result) == 0:
+            return None
+
+        return max(result, key = lambda c: c.condition)
+
+        # condition_check = 0.0
+        # best_in_category = []
+
+        # for item in self.inventory:
+        #     if item.category == category and item.condition > condition_check:
+        #         condition_check = item.condition
+        #         best_in_category.clear()
+        #         best_in_category.append(item)
+
+        # return best_in_category[0]
+
+
+
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        '''
+        takes the best item by category for two vendors and swaps them with each other
+        '''
+        # Helpers
+        my_best_category = self.get_best_by_category(their_priority)
+        their_best_category = other.get_best_by_category(my_priority)
+
+        if len(self.inventory) == 0 or len(other.inventory) == 0 or my_best_category is None or their_best_category is None:
```

Comment:
This conditional only needs to check if the `best_category` items are not None, because the length of the inventory doesn't tell us anything about the number of items they have in that category.
```suggestion
        if my_best_category is None or their_best_category is None:
```